atc/exec/put_inputs.go

```diff
@@ -70,6 +71,45 @@ func (i specificInputs) FindAll(artifacts *artifact.Repository) ([]worker.InputS
 	return inputs, nil
 }
 
+type detectInputs struct {
+	inputs []string
+}
+
+func NewDetectInputs(params map[string]interface{}) PutInputs {
+	var inputs []string
+	for _, value := range params {
+		switch actual := value.(type) {
+		case string:
+			input := actual
+			if idx := strings.IndexByte(actual, '/'); idx >= 0 {
```

Comment:
do we also want to test this functionality, i.e. stripping slashes?

Also is there a reason we do it this way rather than strings.TrimSuffix?

Reply:
I'd also add that it might be worth considering `path/fiilepath#SplitLiist` too 😁  https://golang.org/pkg/path/filepath/#SplitList (as we're dealing with file paths here)

Reply:
@ddadlani It's not stripping slashes here, the idea is to grab the first path segment, i.e. `foo` in `foo/bar/baz`, to identify (guess) the input name.

Reply:
ah ok, thanks. That seems like it's definitely part of the functionality then, since we're coming up with an (expected) identifier. Having a test for it would help give context about that.